sql/item_jsonfunc.cc

```diff
@@ -3911,7 +3911,22 @@ int Arg_comparator::compare_e_json_str_basic(Item *j, Item *s)
 String *Item_func_json_arrayagg::get_str_from_item(Item *i, String *tmp)
 {
   m_tmp_json.length(0);
-  if (append_json_value(&m_tmp_json, i, tmp))
+  if (i->result_type() == STRING_RESULT) {
```

Comment:
small style nit, when using `if .. {` the `{` goes on the new line.

Likewise with `else` below, else starts on the new line.